### PR TITLE
Dashboard Overview components

### PR DESCRIPTION
**Pull request by chrisjgf**
## Stats component

Adds stats component. Option to stake MTA is visible if the user has a non-zero balance.

| XS        | M          |
| --- | --- |
| Screenshot 2020-11-17 at 15 04 24| Screenshot 2020-11-17 at 15 05 40 |

| XL  | XL (NO MTA)
| --- | --- |
| Screenshot 2020-11-17 at 15 05 52 | Screenshot 2020-11-17 at 15 32 04 |

## DAO component

Adds dao component w/ navigation states to learn more. Detail screen includes address, description & balances.

Currently using mock data for tokens.

| Default        | Selected          |
| --- | --- |
| Screenshot 2020-11-19 at 15 32 37 | Screenshot 2020-11-19 at 15 32 43 |

| Row  | Stacked |
| --- | --- |
| Screenshot 2020-11-19 at 15 32 59| Screenshot 2020-11-19 at 15 33 24|

## Overview
| 1440  | 
| --- | 
| Screenshot 2020-11-19 at 16 01 33 |

### CODE REVIEW

File: src/components/pages/Govern/GovernDAO.tsx [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+    <StyledToken key={key}>
+      <Circle color="#eee">
+        <img src={image} alt="mta" />
+      </Circle>
+      <CountUp end={valueRounded} decimals={0} />
+    </StyledToken>
+  );
+};
+
+export const GovernDAO: FC = () => {
```
Comment (JamesLefrere) [inline src/components/pages/Govern/GovernDAO.tsx:263]: Also probably not the best name... what does it do?

File: src/components/pages/Govern/GovernDAO.tsx [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+  const [state, setState] = useState<State>(DEFAULT);
+
+  const handleNavigateClick = useCallback(() => setState(DEFAULT), []);
+  const handleStateSelect = useCallback((s: State) => setState(s), []);
+
+  const { title, address, tooltip } = mapDataToState(state);
+
+  // mocked
+  const tokenMapping: Map<State, TokenProps[]> = new Map([
```
Comment (JamesLefrere) [inline src/components/pages/Govern/GovernDAO.tsx:272]: Could be statically defined

File: src/components/pages/Govern/GovernDAO.tsx [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+      <Header>
+        <div>
+          {!isDefault && (
+            <NavigateButton type="button" onClick={handleNavigateClick} />
+          )}
+          <h3>{title}</h3>
+          {tooltip && <Tooltip tip={tooltip} />}
+        </div>
+        {address && (
+          <Address
```
Comment (JamesLefrere) [inline src/components/pages/Govern/GovernDAO.tsx:296]: Aren't `target` and `rel` already part of that component? They should be either way, sensible defaults

File: src/components/pages/Govern/GovernDAO.tsx [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+            href={getEtherscanLink(address, 'address')}
+          >
+            <span>{truncateAddress(address)}</span>
+          </Address>
+        )}
+      </Header>
+      <Items>
+        {isDefault ? (
+          items.map(s => (
+            <StateSelector state={s} onClick={handleStateSelect} />
```
Comment (JamesLefrere) [inline src/components/pages/Govern/GovernDAO.tsx:308]: Surprised eslint hasn't noticed this needs a key prop :thinking:

File: src/components/pages/Govern/GovernStats.tsx [hunk 5]
```diff
@@ -1,12 +1,180 @@
...
+    stakeData?.simulatedData || {};
+  const { totalValue, totalStakingRewards, lockTimes, votingToken } =
+    incentivisedVotingLockup || {};
+
+  const userHasBalance = (metaToken?.balance.simple ?? 0) > 0.1;
+  const totalSupply = votingToken?.totalSupply;
+  const formattedBalance = metaToken?.balance.simple.toFixed(2);
+  const formattedAPY = simUserStakingReward?.currentAPY?.toFixed(0);
+
+  const lockPeriod = useMemo(() => {
```
Comment (JamesLefrere) [inline src/components/pages/Govern/GovernStats.tsx:149]: No need for useMemo (probably creates more overhead than it saves)

**Review by JamesLefrere — CHANGES_REQUESTED** [review]
Looking great visually, just a few points relating to simplification :+1:

File: src/components/pages/Govern/GovernDAO.tsx (hunk 1 above)
Comment (chrisjgf) [reply src/components/pages/Govern/GovernDAO.tsx:263]: `GovernPublicWalletOverview` | `GovernWalletOverview` ?

File: src/components/pages/Govern/GovernDAO.tsx (hunk 4 above)
Comment (chrisjgf) [reply src/components/pages/Govern/GovernDAO.tsx:308]: Good spot!

File: src/components/pages/Govern/GovernDAO.tsx (hunk 3 above)
Comment (chrisjgf) [reply src/components/pages/Govern/GovernDAO.tsx:292]: This is a different component (styled a link) to the Address component defined elsewhere in the app

File: src/components/pages/Govern/GovernDAO.tsx [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+  background-position-x: 50%;
+  background-position-y: 50%;
+  width: 1.25rem;
+  height: 1.5rem;
+  margin-right: 0.5rem;
+  border: none;
+  cursor: pointer;
+`;
+
+const StateSelector: FC<{
```
[the remark this answers is not in the document]
Comment (chrisjgf) [reply src/components/pages/Govern/GovernDAO.tsx:235]: This changes the UI state of the component - might be a bad design pattern, will shoot you a message

File: src/components/pages/Govern/GovernDAO.tsx (hunk 3 above)
Comment (JamesLefrere) [reply src/components/pages/Govern/GovernDAO.tsx:292]: Ah ok :+1:

File: src/components/pages/Govern/GovernDAO.tsx (hunk 1 above)
Comment (JamesLefrere) [reply src/components/pages/Govern/GovernDAO.tsx:255]: Something like that :) `DaoWalletOverview` maybe?

**Review by JamesLefrere — APPROVED** [review]
Mergy Mergy

**Pull request merged**